contracts/entry-point/src/execute.rs

```diff
@@ -407,6 +420,35 @@ pub fn execute_user_swap(
                 .add_message(user_swap_msg)
                 .add_attribute("action", "dispatch_user_swap_exact_asset_out");
         }
+        Swap::SmartSwapExactAssetIn(swap) => {
```

Comment:
By the time we get here we know that the remaining_asset cannot be less than the swap amounts since we do the subtraction earlier on the route, but what if the remaining_asset is more than the swap amounts? so like (route wants to swap 5asset and 5asset, but they sent the contract 15asset)

Reply:
Don't we transfer the balance back? I'll take a look at the code but I don't fully understand why this would be unique to this swap type